Fix initial issues

elierotenberg opened the pull request:
- Remove ignoreVisibilityHidden option (it would be a bad idea anyway)
- Fix issues raised by @mrtbld
- Incorporate UMD as suggested by @ririkiller
- Simplified general structure

elierotenberg committed:
Fix initial issues

- Remove ignoreVisibilityHidden option (it would be a bad idea anyway)
- Fix issues raised by @mrtbld
- Incorporate UMD as suggested by @ririkiller
- Simplified general structure

elierotenberg committed:
Fix README

mrtbld commented on src/perfTrack.js:
```diff
@@ -1,102 +1,85 @@
+
+  for(var x = 0; x < vendors.length && (!requestAnimationFrame || !cancelAnimationFrame); ++x) {
+    g.requestAnimationFrame = g[vendors[x]+_raf];
+    g.cancelAnimationFrame = g[vendors[x]+_caf] || g[vendors[x]+_altCaf];
```
`g.requestAnimationFrame` → `requestAnimationFrame`
`g.cancelAnimationFrame` → `cancelAnimationFrame`

elierotenberg replied:
:+1:

mrtbld commented on src/perfTrack.js:
```diff
@@ -1,102 +1,85 @@
+    typeof g.performance === 'undefined' ||
+    typeof g.document === 'undefined' ||
+    typeof g.requestAnimationFrame !== 'function' ||
+    typeof g.cancelAnimationFrame !== 'function'
```
`g.requestAnimationFrame` → `requestAnimationFrame`
`g.cancelAnimationFrame` → `cancelAnimationFrame`

elierotenberg replied:
:+1:

mrtbld commented on src/perfTrack.js:
```diff
@@ -1,102 +1,85 @@
-        return stopMonitoring();
+      if(callbackHandle) {
+        cancelAnimationFrame(callbackHandle);
+        callbackHandle = null;
```
`if(initialHandle)` and `if(callbackHandle)` are not necessary. `clearTimeout` and `cancelAnimationFrame` are no-op when the arg isn't a valid handle (including `undefined` and `null`)

elierotenberg replied:
:+1:

mrtbld commented on src/perfTrack.js:
```diff
@@ -1,102 +1,85 @@
-      if(callbackHandler) {
-        return;
-      }
       prevFrame = performance.now();
```
What about never doing this and only relying on the arg of the requestAnimationFrame callback (monitorFrame). The first line of `monitorFrame` would then be:

``` js
var frameDuration = prevFrame ? currFrame - prevFrame : null;
```

or

``` js
var frameDuration = prevFrame ? currFrame - prevFrame : -1;
```

This would remove the need for `performance.now`.

elierotenberg replied:
Unfortunalyte after thourough testing it seems that the parameter passed to the callback of `requestAnimationFrame` is inconsistent accross browsers. We'll therefore use manual timing, but use `Date.now` instead for `performance.now` since we don't need nanosecond accuracy - millisecond accuracy is enough at the framerate timescale (~16.67ms order of magnitude).

elierotenberg committed:
Fix PR issues and pass document.hidden to callback

- Timestamping is now done with Date.now() instead of performance.now() -
  we don't need nanosecond accuracy for framerate measurement (millisecond
  accuracy is enough)
- document.hidden is now simply passed as the 2nd parameter to the
  callback - it is the responsibility of the callback to ignore slow
  frames if document is hidden if they wish so

flagrede commented on src/perfTrack.js:
```diff
@@ -1,102 +1,107 @@
-      }
-      prevFrame = performance.now();
-      callbackHandler = requestAnimationFrame(monitorFrame);
+      initialHandle = null;
```
Is it necessary to set null here, since the variable will receive the setTimeout reference at the end of the function?

flagrede replied:
I got it wrong, actually startMonitoring is called after but then that mean that when stopMonitoring is called we'll be doing clearTimeout(null)?

elierotenberg replied:
`clearTimeout(null)` and `cancelAnimationFrame(null)` are no-ops (they don't throw)
